### PR TITLE
Delete planex-manifest and add metadata to the generated spec file

### PR DESCRIPTION
**Pull request by mseri**
This address both https://github.com/xenserver/planex/issues/546 https://github.com/xenserver/planex/issues/502 based on the discussions that we have been having in the last few days

### CODE REVIEW

File: planex/cmd/makesrpm.py [hunk 1]
```diff
@@ -136,14 +129,20 @@ def populate_working_directory(tmpdir, spec):
...
 
     with open(newspec, "w") as out:
-        out.writelines(spec.rewrite_spec())
+        out.writelines(
+            spec.rewrite_spec(srpm_sources=srpm_sources, manifests=manifests))
 
-    if manifests:
-        add_manifest_entry(manifests, newspec)
-    else:
+    if not manifests:
```
Comment (MarkSymsCtx) [inline planex/cmd/makesrpm.py]: We probably don't need this anymore if we're not strictly limited to reading the SHA from the .gitarchive-info.

File: planex/spec.py [hunk 2]
```diff
@@ -560,10 +560,16 @@ def specpath(self):
...
+    # pylint: disable=too-many-locals
+    def rewrite_spec(self, srpm_sources=None, manifests=None):
         """
         Rewrite the sources and patches in the spec file, also inserting the
-        names of all patchqueue patches into it
+        names of all patchqueue patches into it.
+
+        If [srpm_sources] is a list of sources, it will add comments in the
+        spec file describing the origin of the source.
+        If [manifests] is true it will add `Provides: gitsha(sha) = url` using
```
Comment (MarkSymsCtx) [inline planex/spec.py]: Hopefully this is not what happens, should be `Provides: gitsha(URI) = SHA`

File: planex/spec.py [hunk 3]
```diff
@@ -603,28 +609,70 @@ def sorted_by_key(dictionary):
...
+            if isinstance(pq, GitPatchqueue)
+            else "# Patchqueue: {}\n".format(pq.url)
+            for pq in patchqueues if srpm_sources is not None
+        )
+
+        if manifests is None:
+            manifests = {}
+
+        manifest_metadata = (
+            "Provides: gitsha({}) = {}\n".format(sha, url)
```
Comment (MarkSymsCtx) [inline planex/spec.py]: This is backwards. swap sha and uri

File: planex/spec.py (hunk 2 above)
Comment (mseri) [reply planex/spec.py]: It is :P I'll update the code and fix the docstring

File: planex/spec.py (hunk 3 above)
Comment (mseri) [reply planex/spec.py]: Oops, I'll fix it

**Pull request merged**